### DISCUSSION TITLE
How to change citation behaviour when using Typst

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#10446] [Q&A] [answered]
### Description

(I am sorry I repeated my request from #10440, I forgot I had already submitted)

I am creating a Quarto extension for Typst format where the citations are shown on the margin. For that, I need to be able to allow the user to specify a `dy` parameter to avoid colisions in the margin (see example in the image)
<img width="611" alt="Screenshot 2024-08-01 at 13 55 54" src="https://github.com/user-attachments/assets/8d16bdcf-79e5-4dbb-a650-61a8cc649574">
I imagine that in Quarto `article.qmd` the user will be able to add a place parameter in the citation with something like `In @Tufte2001[7em]` specifying that the citation  should be placed `7em` bellow.

In Typst, I tried to overload the citation with `show cite: margincite` where `margincite` is a function of mine. I noticed that the argument passed then is already a `citation object` with the supplement inside.  I need the parameters of the citation as string not content.

Any idea on how can I change `citation` in a format `Typst`?  I looked on all partial files and couldn't find.  I also couldn't find a `include-before` or `after` that seems to be in the template.

**Comment by mcanouil** [comment]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by fredguth** [reply]
I did create also an issue didn't I.  Sorry.  I sometimes forget that I have already asked in a place and ask in another. 

https://github.com/fredguth/mwe-citation-quarto-typst
Here is the (non) working example.

I hope it is more easy to understand.  The problem is that I need to handle each parameter of citation and I receive in Typst a citation content already.

**Reply by fredguth** [reply]
Can I merge my 2 discussions? #10446

**Reply by gordonwoodhull** [reply, marked as answer]
Continued in #10440
